src/server/zset_family.cc

```diff
@@ -99,6 +101,10 @@ OpResult<PrimeIterator> FindZEntry(const ZParams& zparams, const OpArgs& op_args
   }
 
   PrimeIterator& it = add_res.first;
+  if (!add_res.second && it->second.ObjType() != OBJ_ZSET) {
```

Comment:
but what it `zparams.override` is true and the existing value is not zset?

Reply:
I was under the impression that there's no way to convert a value to a different type without deleting it first, and so I assumed `override` means something else here.
Anyway, reverted, but I'd like to learn more on how this can happen because it might make the current implementation miss such type conversions. Indeed it looks like `zinterstore` and `zunionstore` allow overriding keys with different types (I just checked with redis). Are you aware of any other cases in which this is possible?